src/llvm_jit_context.cpp

```
@@ -426,6 +440,28 @@ llvm_bpf_jit_context::create_and_initialize_lljit_instance()
 	static ExitOnError exitOnErr;
 	// Create a JIT builder
 	SPDLOG_DEBUG("LLVM-JIT: Creating LLJIT instance");
+    // Preload libLLVM before creating LLJIT so that ORC runtime wrapper symbols
+    // (e.g. llvm_orc_registerEHFrameSectionWrapper) are visible during create.
+    // Allow overriding SONAME via environment variable BPFTIME_LLVM_SONAME.
+    {
+        const char *envSoname = ::getenv("BPFTIME_LLVM_SONAME");
+        const char *candidates[] = {
+            envSoname && envSoname[0] ? envSoname : (const char *)nullptr,
+            "libLLVM-17.so",
+            "libLLVM.so",
+            "libLLVM-17.0.6.so",
+        };
+        for (const char *name : candidates) {
+            if (!name)
+                continue;
+            auto ok = llvm::sys::DynamicLibrary::LoadLibraryPermanently(name);
+            (void)ok;
```

We should print logs when failed instead of ignoring it